controllers/helmrepository_controller_oci.go

```diff
@@ -327,30 +307,40 @@ func (r *HelmRepositoryOCIReconciler) validateSource(ctx context.Context, obj *s
 
 	chartRepo, err := repository.NewOCIChartRepository(obj.Spec.URL, repository.WithOCIRegistryClient(registryClient))
 	if err != nil {
-		e := &serror.Stalling{
-			Err:    fmt.Errorf("failed to parse URL '%s': %w", obj.Spec.URL, err),
-			Reason: sourcev1.URLInvalidReason,
-		}
-		conditions.MarkFalse(obj, meta.ReadyCondition, e.Reason, e.Err.Error())
-		return sreconcile.ResultEmpty, e
+		e := fmt.Errorf("failed to parse URL '%s': %w", obj.Spec.URL, err)
+		conditions.MarkStalled(obj, sourcev1.URLInvalidReason, e.Error())
+		conditions.MarkFalse(obj, meta.ReadyCondition, sourcev1.URLInvalidReason, e.Error())
+		result, retErr = ctrl.Result{}, nil
+		return
 	}
+	conditions.Delete(obj, meta.StalledCondition)
```

Comment:
we already delete the `StalledCondition` after checking if we have an OCI url, do we need to do it once more?

Reply:
Based on the context, no, it's not needed.
I added it because of the habit of ensuring a condition is deleted on success when a condition is added on failure.